Pause control does not work in in-window mode

danae404 opened the pull request:
#### c67d585912b331ac7757866a1f969e25de04553c
<pre>
Pause control does not work in in-window mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=269951">https://bugs.webkit.org/show_bug.cgi?id=269951</a>
<a href="https://rdar.apple.com/122834226">rdar://122834226</a>

Reviewed by Jer Noble.

This bug was caused by both our event handler for clicking on the pause button
and youtube&apos;s event handler for clicking on the video both being activated,
causing the video to pause and then immediately be played again. To prevent
this, in our button&apos;s event handler we now call stopPropagation().

* Source/WebCore/Modules/modern-media-controls/controls/button.js:
(Button.prototype.handleEvent):

Canonical link: <a href="https://commits.webkit.org/275229@main">https://commits.webkit.org/275229@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9cd7995a71e0f1eca71bce5d3010748a418a66e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41184 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20249 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43747 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37276 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43491 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23307 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34103 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41758 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17168 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35468 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14724 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14901 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36470 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45068 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37416 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36787 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40530 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15999 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13116 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38910 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17618 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9251 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17670 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17262 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/Modules/modern-media-controls/controls/button.js:
```diff
@@ -150,6 +150,7 @@ class Button extends LayoutItem
 
     handleEvent(event)
     {
+        event.stopPropagation();
```
should we limit this to just the pause event ?

jernoble replied:
This would only be events that directly targeted this button, like "click", "mouseover", etc.

danae404 committed:
Pause control does not work in in-window mode
https://bugs.webkit.org/show_bug.cgi?id=269951
rdar://122834226

Reviewed by Jer Noble.

This bug was caused by both our event handler for clicking on the pause button
and youtube's event handler for clicking on the video both being activated,
causing the video to pause and then immediately be played again. To prevent
this, in our button's event handler we now call stopPropagation().

* Source/WebCore/Modules/modern-media-controls/controls/button.js:
(Button.prototype.handleEvent):

Canonical link: https://commits.webkit.org/275229@main
